### DISCUSSION TITLE
Is it possible to build a website from multiple repos?

### DISCUSSION THREAD

**Discussion by pat-alt** [quarto-dev/quarto-cli#1282] [Q&A] [answered]
This might be a silly question/idea, but anyway here goes:

I have [blog](https://www.paltmeyer.com/blog/) built with Quarto and served from this [repo](https://github.com/pat-alt/blog). Recently, I have also moved my personal [website](https://github.com/pat-alt/pat-alt.github.io) to Quarto, which is served from my root Github Pages [repo](https://github.com/pat-alt/pat-alt.github.io). I like having separate repos for both sites, but would ideally like to serve the blog as part of my website (like [Quarto's blog](https://quarto.org/docs/blog/)). So my questions is as follows:

**Is it possible to build the `/contents/blog/` section of my website from the separate blog repo?**

I guess I could work with git submodules, but have always found them a bit fiddly. Any advice would be much appreciated.

**Comment by jjallaire** [comment, marked as answer]
Submodules is certainly one way to go (but I agree they are too fiddly)

Another would be to implement a shared (i.e. identical looking) Navbar across the site and the blog (this would require two separate publish operations)

Yet another would be to use a post-render script on the website (https://quarto.org/docs/projects/quarto-projects.html#pre-and-post-render-scripts) to copy the blog into the `_site` directory (note you'd probably only want to do the copy when `QUARTO_PROJECT_RENDER_ALL="1"` so it doesn't happen for incremental preview).

**Reply by pat-alt** [reply]
Thanks 👍🏽 will probably go with the pre/post-render

**Reply by adrien-perello** [reply]
I have the same use case (personal website that showcases different projects, each project having its own repository).
I am struggling with the [pre and post render scripts](https://quarto.org/docs/projects/scripts.html#pre-and-post-render). Anyone would have an example that I could get inspiration from? @pat-alt maybe?

**Reply by pat-alt** [reply]
I had a quick look at post-render, but had some trouble as well. Will try to remember to post here if/when I sort it out 👍

**Reply by adrien-perello** [reply]
Thanks for the quick reply @pat-alt , anyone else reading this thread then maybe? 🙏

**Reply by mcanouil** [reply]
The idea seems similar to another thread about having revealjs and a website from the same content.
You have to write a script/function that move around the different parts.

https://github.com/quarto-dev/quarto-cli/discussions/1433#discussioncomment-3220454

**Reply by andrewpbray** [reply]
I had a similar situation a few months back. I had one repo that held my [course materials](https://github.com/stat20/course-materials) and a separate repo that held the [course website](https://github.com/stat20/website). I wanted the experience of authoring materials to be a focused on content (slides and learning outcomes) and I wanted to the website, once set up, to just work and suck in the content. I didn't combine the two in one repo because it's a blogdown site and I didn't want all that source to muddy up the course materials. I looked into the submodules thing but was warned away by people wiser in git than I.

I ended up using a [github action](https://github.com/stat20/course-materials/blob/master/.github/workflows/sync-w-website.yaml) that copied slides from the course materials repo into the website. At first the idea of copying and losing commit history gave me the willies but it worked fine and the commit history is still preserved over in the course materials. The workflow was super simple: 1) finish up slides in clean course materials repo 2) push commits 3) run to class! The structure of the blog also made it so that the new files were automatically added into the nav when the site rendered.

**Reply by mcanouil** [reply]
Depending on the use cases, I do not mind much using git submodule.
For example, I created few Quarto templates, initially in one repository, but it's currently not possible to install a subdirectory with Quarto.
To have access to them all still in one place, I used submodule and GHA to update those weekly.

https://github.com/mcanouil/publishing-templates

**Comment by MitchellAcoustics** [comment]
Does anyone know whether there is a better way to achieve this now? For my use case, I build my [personal academic website](https://drandrewmitchell.com/research/papers.html) with quarto and one section is 'Open Source Papers' where I publish quarto versions of my open access papers, code and all. Since quarto renders all from the same virtual environment, this gets unmanageable for each new paper, since I use new and updated versions of packages in new papers. 

I'd like to be able to have a separate repo for each paper (which is also included in the journal version when published) which renders independently of my website, then can just be pulled in and integrated as a page on the website.

Any ideas of the best way to handle this?

**Reply by mcanouil** [reply]
See previous answers.
Nothing changed and unlikely to change as one of the design choice of Quarto is to have project self-contained.
